packages/blade/src/components/Button/BaseButton/buttonTokens.ts

```diff
@@ -0,0 +1,102 @@
+import type { ThemeTokens } from '../../../tokens/theme/theme.d';
+import type { TypographyPlatforms } from '../../../tokens/global/typography';
+import type { Required } from '../../../_helpers/types';
+import type { IconProps } from '../../Icons';
+import type { BaseButtonProps } from './BaseButton';
+
+export type ButtonMinHeight = 48 | 40 | 32 | 28;
+
+export type ButtonTypography = {
+  [key in TypographyPlatforms]: {
+    fonts: {
+      size: Record<
+        Required<BaseButtonProps['size']>,
+        keyof ThemeTokens['typography'][key]['fonts']['size']
+      >;
+    };
+    lineHeights: Record<
+      Required<BaseButtonProps['size']>,
```

Comment:
Removed `undefined` & `null` from the type in cases where we want to enforce the type while ensuring that the value exists. Size by default can be undefined in the props but since we assign it a default value as `medium`, we want to remove `undefined` from the type here

Reply:
You can remove the Required type (it also clashes with the built-in Required type). 

and replace with the built-in [`NonNullable`](https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype) type. It does the same thing.